src/tasks/Publish/get-npm-client.js

```diff
@@ -1,24 +1,8 @@
-import execa from 'execa';
-
-let cacheNpmClient = null;
+import getYarnLineage from './get-yarn-lineage';
 
 export default async function getNpmClient() {
-  if (cacheNpmClient) {
-    return cacheNpmClient;
-  }
-
-  const useYarn = await shouldUseYarn();
+  const useYarn = await Boolean(getYarnLineage());
```

Comment:
This should be:
```suggestion
  const useYarn = Boolean(await getYarnLineage());
```

otherwise it will always be true?
<img width="305" alt="Screen Shot 2022-06-17 at 17 23 09" src="https://user-images.githubusercontent.com/7464663/174396384-6dc358ff-4acf-4141-909c-cd5f72dc0422.png">

Reply:
Good catch!  Fixed (and added a test).